src/main/java/hudson/plugins/tics/AbstractApiCall.java

```diff
@@ -101,6 +98,9 @@ protected final CloseableHttpClient createHttpClient() throws KeyManagementExcep
 
     protected boolean isProxyExempted(final String urlStr, final List<Pattern> noProxyPatterns) {
         Matcher matcher;
+        // Bypassing proxy for internal hosts by default
+        noProxyPatterns.add(Pattern.compile("localhost"));
+        noProxyPatterns.add(Pattern.compile("127\\..*"));
```

Comment:
It is not a good idea to modify the incoming list. Callers of this method might not expect their input argument to be modified. In fact, it seems that you are modifying the list that comes out of 'proxy.getNoProxyHostPatterns()'. This method is offered by Jenkins API, so modifying it might be dangerous and might throw an exception (although apparently it does not because you tested it).

Also, in the test, you call `isProxyExempted` on the list in the test case, which modify the list in the test case. It works in this case because you use each test case only once, but it not nice: the test cases should be immutable.

I would advise you to use Guava's ImmutableList more. The parameter here should be `ImmutableList<Pattern>` so that callers can be sure that their input arguments are not modified. In this method you can then iterate over the concatenated list as follows: `Iterables.concat(noProxyPatterns, ImmutableList.of(Pattern.compile("localhost"), Pattern.compile("127\\..*")))`. You can also factor out the localhost patterns list in a constant, i.e.:

`public static final ImmutableList<Pattern> LOCALHOST_PATTERNS = ImmutableList.of(Pattern.compile("localhost"), Pattern.compile("127\\..*")));`

Reply:
Fixed in https://github.com/jenkinsci/tics-plugin/pull/28/commits/a66271079d7c5abd7e61140174044efb021cd579